src/main/java/it/gov/pagopa/pu/workflow/wf/classification/iuf/classification/IufClassificationWFImpl.java

```diff
@@ -0,0 +1,136 @@
+
+      // signal transfer classification for each transfer
+      uniqueTransfers.forEach(transfer2ClassifyDTO -> {
+
+        String iuv = transfer2ClassifyDTO.getIuv();
+        String iur = transfer2ClassifyDTO.getIur();
+        int transferIndex = transfer2ClassifyDTO.getTransferIndex();
+
+        startTransferClassificationActivity.signalTransferClassificationWithStart(result.getOrganizationId(), iuv, iur, transferIndex);
+      });
+
+    } else {
+      log.warn("Result is null or classification was not successful for organizationId {}",
+        result != null ? result.getOrganizationId() : "N/A");
+    }
+
+  }
+
+
+  @Override
+  public void notifyTreasury(IufClassificationNotifyTreasurySignalDTO signalDTO) {
+
+    log.info("Handling treasury notification in iuf classification: {}", signalDTO);
+    boolean clearedResult = clearClassifyIufActivity.deleteClassificationByIuf(
+      signalDTO.getOrganizationId(),
+      signalDTO.getIuf());
+
+    log.info("IUF receipt classification cleared with result {} for {}", clearedResult, signalDTO);
+
+    IufClassificationActivityResult iufClassificationActivityResult = iufClassificationActivity.classify(signalDTO.getOrganizationId(), signalDTO.getTreasuryId(), signalDTO.getIuf());
+
+    if (iufClassificationActivityResult.isSuccess()) {
+      if (result == null) {
+        result = iufClassificationActivityResult;
+      } else {
+        Set<Transfer2ClassifyDTO> uniqueTransfers = new HashSet<>(result.getTransfers2classify());
+        uniqueTransfers.addAll(iufClassificationActivityResult.getTransfers2classify());
+        List<Transfer2ClassifyDTO> listTransfers = uniqueTransfers.stream().toList();
+        result.setTransfers2classify(listTransfers);
+
+      }
+    } else {
+      log.error("Error in treasury classification for organizationId {}", signalDTO.getOrganizationId());
+    }
+  }
+
+  @Override
+  public void notifyPaymentsReporting(IufClassificationNotifyPaymentsReportingSignalDTO signalDTO) {
+
+    log.info("Handling payments reporting notification in iuf classification: {}", signalDTO);
+    boolean clearedResult = clearClassifyIufActivity.deleteClassificationByIuf(
+      signalDTO.getOrganizationId(),
+      signalDTO.getIuf());
+
+    log.info("IUF receipt classification cleared with result {} for {}", clearedResult, signalDTO);
+
+    if (result == null) {
+      result = IufClassificationActivityResult.builder()
+        .organizationId(signalDTO.getOrganizationId())
+        .success(true)
+        .transfers2classify(signalDTO.getTransfers2classify())
+        .build();
+    } else {
+
+      Set<Transfer2ClassifyDTO> uniqueTransfers = new HashSet<>(result.getTransfers2classify());
+      uniqueTransfers.addAll(signalDTO.getTransfers2classify());
+
+      List<Transfer2ClassifyDTO> listTransfers = uniqueTransfers.stream().toList();
+      result.setTransfers2classify(listTransfers);
+    }
```

Comment:
just add the request 2 notify
```suggestion
    toNotify.add(IufClassificationActivityResult.builder()
        .organizationId(signalDTO.getOrganizationId())
        .success(true)
        .transfers2classify(signalDTO.getTransfers2classify())
        .build());
```

Reply:
OK